resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -224,6 +223,7 @@ public DnsNameResolver(
      * @param decodeIdn {@code true} if domain / host names should be decoded to unicode when received.
      *                        See <a href="https://tools.ietf.org/html/rfc3492">rfc3492</a>.
      */
+    @Deprecated
```

Comment:
this class is marked as unstable. my preference would be to avoid adding deprecated methods if possible. also most users will likely use the builder to construct this object always.